charts/pega/templates/_helpers.tpl

```diff
@@ -164,6 +164,19 @@ until cqlsh -u {{ $cassandraUser | quote }} -p {{ $cassandraPassword | quote }}
   {{- add $passivationTime $passivationDelay -}}
 {{- end -}}
 
+# Determine application root context to use in pega tomcat nodes
+{{- define "pega.applicationContextPath" -}}
+   {{- if .node.ingress -}}
+      {{- if .node.ingress.appContextPath -}}
+         {{ .node.ingress.appContextPath }}
+      {{- else -}}
+         prweb
+      {{- end -}}	 
+   {{- else -}}
+      prweb
+   {{- end -}}
```

Comment:
Would it be possible to use an `and` here, or would that fail if there's no `.node.ingress`?
I'm thinking possibly something like this.
```suggestion
   {{- if and .node.ingress .node.ingress.appContextPath -}}
      {{ .node.ingress.appContextPath }}
   {{- else -}}
      prweb
   {{- end -}}
```

Reply:
@pega-roska I agree that this is more concise, but helm seems to be evaluating .node.ingress.appContextPath regardless of the evaluation of .node.ingress to false.  I think I will leave this as-is.

Reply:
Yea, Helm seems to be strange in that way.  I've banged my head against it in the past.  The only other workaround I have found to use is to set a local variable which will default to an object rather than _null_ so you can query it without failure.  See [the health probes](https://github.com/pegasystems/pega-helm-charts/blob/master/charts/pega/templates/_pega-deployment.tpl#L156) for an example.